python/cudf/cudf/pandas/__init__.py

```diff
@@ -18,6 +18,25 @@ def install():
     loader = ModuleAccelerator.install("pandas", "cudf", "pandas")
     global LOADED
     LOADED = loader is not None
+    import os
+
+    cudf_pandas_mr = os.getenv("CUDF_PANDAS_MEMORY_RESOURCE", None)
+    if cudf_pandas_mr is not None:
+        import rmm
+        import rmm.mr
+
+        cudf_pandas_mr = getattr(rmm.mr, cudf_pandas_mr, None)
+        if cudf_pandas_mr is not None:
+            from rmm.mr import PoolMemoryResource
+
+            mr = PoolMemoryResource(
```

Comment:
Do we _always_ want the user's specified MR to be wrapped in a pool?

Reply:
I don't know, open to suggestions here. Or we can change this as we know which works the best from experimenting.

Reply:
Certainly not. In the example in the PR description you specify CudeAsyncMemoryResource. That already has a pool, you probably wouldn't always want to wrap that in an RMM PoolMemoryResource.

I thought in the meeting we discussed just providing a limited set of choices here. More sophisticated users should be able to import RMM first and configure the current device resource to override if they have very specific MR needs.

Reply:
We do a similar thing in libcudf NVBench benchmarks. Here are the options we expose there, for reference: https://github.com/rapidsai/cudf/blob/8cc4cc16fc6538f706b9f795d0879bdd0ba442a1/cpp/benchmarks/fixture/nvbench_fixture.hpp#L73-L84

Reply:
@harrism Do you feel comfortable having a similar option set for `cudf.pandas`?

Reply:
Do you think we need arena?

Reply:
> Do you think we need arena?

Maybe, I haven't tried it, but its fragmentation avoidance might be useful. 
However, to limit the user complexity/documentation of cudf.pandas, I am also fine skipping `arena`. We can always add it later if it turns out to be useful.

Reply:
I think you can also manually enable it by importing RMM and enabling it before importing cudf.pandas.

Reply:
I updated the PR to have simpler rmm mode selection and added pytests for the same.

@harrism @bdice This should be ready for review.